code/utils.py

```diff
@@ -7,6 +7,7 @@
 """
 
 
+from six.moves import xrange
```

Comment:
I think it will be a better example to follow if we don't use this import and change everywhere xrange to range. Can you do that?